crates/sui-types/src/transaction.rs

```diff
@@ -868,12 +868,13 @@ pub fn write_sep<T: Display>(
     items: impl IntoIterator<Item = T>,
     sep: &str,
 ) -> std::fmt::Result {
-    let mut xs = items.into_iter().peekable();
-    while let Some(x) = xs.next() {
-        write!(f, "{x}")?;
-        if xs.peek().is_some() {
-            write!(f, "{sep}")?;
-        }
+    let mut xs = items.into_iter();
+    let Some(x) = xs.next() else {
+        return Ok(());
+    };
+    write!(f, "{x}")?;
+    for x in xs {
+        write!(f, "{sep}{x}")?;
```

Comment:
Improvement as per @amnn's suggestion in a [previous PR](https://github.com/MystenLabs/sui/pull/15719/files#r1455324668).